consensus/reactor_test.go

```diff
@@ -65,6 +65,12 @@ func TestValidatorSetChanges(t *testing.T) {
 	nPeers := 8
 	nVals := 4
 	css := randConsensusNetWithPeers(nVals, nPeers, "consensus_val_set_changes_test", newMockTickerFunc(true))
+
+	// otherwise, the test is failing with timeout error
```

Comment:
the commits are nil unless explicitly received otherwise. i think for the tests its better to not use the timeout_commit, and to wait for all the votes, because otherwise we can end up with timing dependencies in the testing code which can lead to nondeterministic failures. that was part of the reason for this change originally. but for live chains we want to be able to use timeout_commit. so maybe the way to test it is to start a 4 validator testnet but with only 3 validators online and have them all send votes and then after timeout_commit theyll move to the next height